torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -17,6 +27,66 @@
 DETECTOR_IS_POST_OBS_KEY = "is_post_observer"
 DETECTOR_OBS_ARGS_KEY = "observer_args"
 
+# Mapping related code
+class DetectorQConfigInfo():
+    r"""
+    This class contains the QConfig information for a single module.
+    The list of variables / values this contains can grow depending on the
+    extensibility of the qconfig mapping feature set but this currently includes:
+    - if activation observer is dynamic
+    - if weight observer is per channel
+
+
+    Args:
+        module_fqn (str): The fully qualified name (fqn) of the module that this
+            information contains info relavent to qconfig for
+    """
+
+    def __init__(self, module_fqn: str):
+        super().__init__()
+        self.module_fqn = module_fqn
+
+        # populate this section with all the variables we might find important
+        # change from none if your detector is actually using this
+        self.is_activation_dynamic = False
+        self.is_weight_per_channel = False
+
+    def generate_qconfig(self, module: torch.nn.Module) -> QConfig:
+        r"""
+        Args:
+            module (torch.nn.Module) The module we are generating
+            the qconfig for
+
+        Returns the generated QConfig according to what a valid configuration is
+        """
+        # Apply suggestions to new qconfig
+        module_qconfig = default_qconfig
```

Comment:
should the default be `default_qconfig` or None?

Reply:
should be default_qconfig because we want to use some level of quantizatio

Reply:
but using quantization is not always good right? e.g. if detector recommends to use per channel quant based on data, but it's not supported on the operator, then using per tensor quant means it will have poor accuracy, is this something we want to recommend?

Reply:
the detector only recommends to use per channel quantization for scenarios in which both per channel and per tensor are supported by the operator, (it cross checks with list of supported modules)

so we won't run into the case where we recommend per channel quant based on data but it's not supported by operator